src/AzureAuth/CommandMain.cs

```diff
@@ -444,6 +494,21 @@ private int GetToken()
             return 0;
         }
 
+        private void SendAuthFlowTelemetryEvents(AuthFlowResult[] results)
+        {
+            if (results != null)
+            {
+                Parallel.ForEach(results, result =>
+                {
+                    var eventData = this.AuthFlowEventData(result);
+                    if (eventData != null)
+                    {
+                        this.telemetryService.SendEvent($"authflow_{result.AuthFlowName}", eventData);

```

Comment:
Since we are using parallel foreach here, let's ensure that `SendEvent` is thread safe. For instance we have 2 threads simultaneously tried to send events then both of them should be able to. It shouldn't be that one succeeds and the other fails. It will result in loss of events in your telemetry.

Reply:
That's a good question, @shalinikhare27. I think they are thread-safe. Because when this method is called, Lasso asynchronously sends event in a separate new process. @kyle-rader please correct me if I'm wrong.

Reply:
The only part that may have thread safety concerns is the logger instance, though I believe NLog loggers are generally thread safe.